spec/package_storage_spec.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+RSpec.describe PackageStorage do
+  FakeBag = Struct.new(:storage_location) do
+    def format
+      "bag"
+    end
+  end
+
+  FakeZip = Struct.new(:storage_location) do
```

Comment:
I _think_ I'm OK with this here -- I think it makes the tests easier to read in comparison with using e.g. `format1` and `format2` or random words instead of `bag` and `zip`. The concern would be the appearance that the spec has some reliance or specification of actual bag or zip behavior, but in fact the actual names of the formats don't matter at all.


, although calling it a `FakeBag` might still imply there's some binding to some of the real bag handling machinery in some sense. I think the alternative would be calling the formats things like `format_one` and `format_two` here, but I think the tests are easier to read as is.

Reply:
Agreed... I'm kind of divided on how realistic the names should be in this one. I think it might become more clear as we think about the distinctions between artifacts, revisions, and the storage formats. For now, I think it reads well enough to make the behavior understandable.